gammapy/datasets/map.py

```diff
@@ -1525,9 +1556,31 @@ def excess(self):
         """Excess (counts - alpha * counts_off)"""
         return self.counts - self.counts_off_normalised
 
+    def npred_sig(self, model=None):
+        """"Model predicted signal counts. If a model is passed, predicted counts from that component is returned.
+            Else, the total signal counts are returned.
+
+            Parameters
+            -------------
+            model: `~gammapy.modeling.models.Models`, optional
+               The model to computed the npred for.
+
+            Returns
+            ----------
+            npred_sig: `gammapy.maps.WcsNDMap`
+        """
+        if model is None:
```

Comment:
I think this can be simplified, as the `if model is None` handling is already implemented in the `MapDataset` class.

Reply:
The behaviour is a bit different in the 2 cases; for `model=None`, `MapDatasetOnOff.npred_sig` is the sum of all model components, whereas for `MapDataset.npred_sig` is `npred - background`.
Do you have a specific suggestion for simplification?

Reply:
Isn't this the same?

Reply:
Ah, I see result-wise this is the same, but implementation-wise it is not. Ok, to keep. Maybe we can simplify this later....